minecraft/realms/realms.go

```diff
@@ -0,0 +1,213 @@
+package realms
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/sandertv/gophertunnel/minecraft/auth"
+	"golang.org/x/oauth2"
+)
+
+// RealmsApi is an instance of the realms api with a token
+type RealmsApi struct {
+	token_src  oauth2.TokenSource
+	xbox_token *auth.XBLToken
+}
+
+// NewRealmsApi returns a new RealmsApi instance with the supplied token source for authentication
+func NewRealmsApi(src oauth2.TokenSource) *RealmsApi {
+	return &RealmsApi{
+		token_src: src,
+	}
+}
+
+// RealmPlayer in a realm returned from the api
+type RealmPlayer struct {
```

Comment:
This package is already called `realms`, so the full identifier of this is `realms.RealmPlayer` now. I suggest renaming this to `Player` to avoid stuttering.